include/extractor/guidance/validation_handler.hpp

```diff
@@ -177,6 +179,81 @@ class ValidationHandler final : public IntersectionHandler
             }
         }
     }
+
+    void checkForSharpDualCarriagewayUturns(const NodeID from_nid,
+                                            const EdgeID via_eid,
+                                            const Intersection &intersection) const
+    {
+        // Detect situations where dual-carriageway fans-in from two ways into one and
+        // the Uturn from a dual-carriageway onto the dual-carriageway should be possible.
+        //
+        // . .
+        //     > .
+        // . .
+        //
+
+        // Todo: are there dual-carriageway intersections of larger size?
+        if (intersection.size() != 3)
+        {
+            return;
+        }
+
+        const auto &via_road = intersection.getUTurnRoad();
+        const auto &leftmost_road = intersection.getLeftmostRoad();
+        const auto &rightmost_road = intersection.getRightmostRoad();
+
+        const auto &via_road_data = node_based_graph.GetEdgeData(via_eid);
+        const auto &leftmost_road_data = node_based_graph.GetEdgeData(leftmost_road.eid);
+        const auto &rightmost_road_data = node_based_graph.GetEdgeData(rightmost_road.eid);
+
+        if (via_road_data.name_id == EMPTY_NAMEID || leftmost_road_data.name_id == EMPTY_NAMEID ||
```

Comment:
this Entire part feels like code-duplication. Wouldn't this offer itself to check for `isSharpTurnOntoDual` and simply compare via/left, via/right for themselves instead of carrying all the left/right things?